lib/resty/kafka/producer.lua

```diff
@@ -353,12 +355,82 @@ local function negotiate_api_version(supported_version_map, api_key, max_support
     return min_version
 end
 
+-- if new arguments already exist in cached producers, then skip creating
+-- may subject to plugin (e.g. "kafka-log") schema updates
+local function _same_cluster(broker_list, producer_config, cluster_name)
```

Comment:
What about removing unwanted keys from the config and comparing hashes?

Reply:
Do you mean compute the hash of tables? Is there any reference on this?

The keys are spread across different objects (e.g. producer, ringbuffer, client, etc).

Reply:
I don't think "comparing table hash" is a good idea. We use that for config sync. It's slow and complex.